tests/effects/tdiagnostic_messages.nim

```diff
@@ -0,0 +1,32 @@
+discard """
+  errormsg: "\'a\' can have side effects"
+  nimout: '''  tdiagnostic_messages.nim(32, 30) Hint: 'a' calls 'callWithSideEffects' which has side effects
+    tdiagnostic_messages.nim(24, 6) Hint: 'callWithSideEffects' called by a `noSideEffect` routine, but has side effects
+      tdiagnostic_messages.nim(26, 34) Hint: 'callWithSideEffects' calls 'indirectCallViaVarParam' which has side effects
+        tdiagnostic_messages.nim(20, 6) Hint: 'indirectCallViaVarParam' called by a `noSideEffect` routine, but has side effects
+          tdiagnostic_messages.nim(21, 7) Hint: 'indirectCallViaVarParam' calls a routine with side effects via hidden pointer indirection
+      tdiagnostic_messages.nim(27, 33) Hint: 'callWithSideEffects' calls 'indirectCallViaPointer' which has side effects
+        tdiagnostic_messages.nim(22, 6) Hint: 'indirectCallViaPointer' called by a `noSideEffect` routine, but has side effects
+          tdiagnostic_messages.nim(23, 32) Hint: 'indirectCallViaPointer' calls a routine with side effects via pointer indirection
+      tdiagnostic_messages.nim(28, 8) Hint: 'callWithSideEffects' calls 'echo' which has side effects
+        ../../lib/system.nim(2004, 6) Hint: 'echo' called by a `noSideEffect` routine, but has side effects
+      tdiagnostic_messages.nim(29, 3) Hint: 'callWithSideEffects' accesses global state 'globalVar'
+        tdiagnostic_messages.nim(19, 5) Hint: 'globalVar' accessed by a `noSideEffect` routine'''
+  file: "tdiagnostic_messages.nim"
+  line: 31
+"""
+
```

Comment:
> The terminal rendering's structure gives context to the hints about what noSideEffect routine is trying to use routines with side effects.
The VSCode hints don't provide that context. Should I change them to someting like:
Hint: 'indirectCallViaVarParam' called by `a`

(i prefer threading so i moved your question here)

i don't use VSCode, can you explain further? change to
```
Hint: 'indirectCallViaVarParam' called by `a`
```
from what?

Reply:
The VSCode screenshot above shows the message ```'globalVar' accessed by a `noSideEffect` routine```.
It seems difficult for a user to answer which `.noSideEffect` routine is trying to use it.
That information is available in the terminal because of the tree structure.

Reply:
got it.
```
Hint: 'callWithSideEffects' called by a `noSideEffect` routine, but has side effects
```
=>
```
Hint: `.sideEffect` 'callWithSideEffects' called by `a`
```

i think is crips and has all the information